src/Helpers/LeanCode.UserIdExtractors/ServiceProviderExtensions.cs

```diff
@@ -0,0 +1,87 @@
+using System.Reflection;
+using LeanCode.DomainModels.Ids;
+using LeanCode.UserIdExtractors.Extractors;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace LeanCode.UserIdExtractors;
+
+public static class UserServiceProviderExtensions
```

Comment:
Oh boi. You did a lot of work, except that, I think, it won't ever be used.
Let's estimate how many "user ids" we have in a project. I would say, that typically, there is only one user id - you either use `Guid` or use one, single typed id like `UserId`. You then use it across the whole system. This means that you need exactly one type `IUserIdExtractor`. What is more, you don't need any assembly scanning at all, because you know the type up-front.